### PR TITLE
Change shutdown_test.cc to use credential provider API to test against  different credential types

### PR DESCRIPTION
**Pull request by danzh2010**

### CODE REVIEW

**Comment by yang-g** [comment]
jenkins: this is ok to test

File: test/cpp/end2end/shutdown_test.cc [hunk 1]
```diff
@@ -83,7 +84,8 @@ class ShutdownTest : public ::testing::Test {
     grpc::string server_address = "localhost:" + to_string(port);
 
     ServerBuilder builder;
-    builder.AddListeningPort(server_address, InsecureServerCredentials());
+    auto server_creds = GetServerCredentials(GetParam());
```
Comment (yang-g) [inline test/cpp/end2end/shutdown_test.cc]: Is this supposed to be `GetCredentialsProvider()->GetServerCredentials`?

Comment (danzh2010) [inline test/cpp/end2end/shutdown_test.cc]: Yes, my local repo needs sync. Added GetCredentialsProvider() to all test_credentials_provider api. PTAL!

**Comment by danzh2010** [comment]
sync with upstream and run clang format. PTAL!

**Comment by danzh2010** [comment]
Done. Please merge this commit. Thanks!

**Comment by yang-g** [comment]
Known test failures: #9497 #9150
merging

**Pull request merged**